src/states/node.rs

```diff
@@ -2203,6 +2229,44 @@ impl Node {
                                error);
                     }
                 }
+
+                if let Some((_sender_prefix, merge_prefix, sections)) =
+                    self.peer_mgr.should_merge() {
+                    let senders = self.out_of_seq_own_merge
+                        .get(&(merge_prefix, sections.clone()))
+                        .map_or(BTreeSet::new(), |senders| if senders.len() == 2 {
+                            senders.clone()
+                        } else {
+                            BTreeSet::new()
+                        });
```

Comment:
If `senders` exists and has length `!= 2`, this allocates two new, empty `BTreeSets` just do do nothing. Maybe:
```rust
if let Some(senders) = self.out_of_seq_own_merge
                        .get(&(merge_prefix, sections.clone()))
                        .iter()
                        .filter(|senders| senders.len() == 2)
                        .cloned()
                        .next() {
```